activation/atxdb.go

```diff
@@ -746,38 +747,38 @@ func (db *DB) ValidateSignedAtx(pubKey signing.PublicKey, signedAtx *types.Activ
 }
 
 // HandleGossipAtx handles the atx gossip data channel
-func (db *DB) HandleGossipAtx(data service.GossipMessage, syncer service.Fetcher) {
+func (db *DB) HandleGossipAtx(ctx context.Context, data service.GossipMessage, syncer service.Fetcher) {
 	if data == nil {
 		return
 	}
-	err := db.HandleAtxData(data.Bytes(), syncer)
+	err := db.HandleAtxData(ctx, data.Bytes(), syncer)
 	if err != nil {
-		db.log.With().Error("error handling atx data", log.Err(err))
+		db.log.WithContext(ctx).With().Error("error handling atx data", log.Err(err))
 		return
 	}
-	data.ReportValidation(AtxProtocol)
+	data.ReportValidation(ctx, AtxProtocol)
 }
 
 // HandleAtxData handles atxs received either by gossip or sync
-func (db *DB) HandleAtxData(data []byte, syncer service.Fetcher) error {
+func (db *DB) HandleAtxData(ctx context.Context, data []byte, syncer service.Fetcher) error {
+	logger := db.log.WithContext(ctx)
 	atx, err := types.BytesToAtx(data)
 	if err != nil {
 		return fmt.Errorf("cannot parse incoming atx")
 	}
 	atx.CalcAndSetID()
 
-	db.log.With().Info("got new atx", atx.Fields(len(data))...)
+	logger.With().Info("got new atx", atx.Fields(len(data))...)
 
 	//todo fetch from neighbour (#1925)
 	if atx.Nipst == nil {
-		db.log.Panic("nil nipst in gossip")
+		logger.Panic("nil nipst in gossip")
 		return fmt.Errorf("nil nipst in gossip")
 	}
 
-	if err := syncer.GetPoetProof(atx.GetPoetProofRef()); err != nil {
+	if err := syncer.GetPoetProof(context.TODO(), atx.GetPoetProofRef()); err != nil {
```

Comment:
```suggestion
	if err := syncer.GetPoetProof(ctx, atx.GetPoetProofRef()); err != nil {
```

Or am I missing something?

If I am, I guess it should still not be `TODO()`

Reply:
already caught and fixed this